src/Propel/Generator/Model/Table.php

```diff
@@ -819,15 +820,39 @@ public function getCrossFks()
         $crossFks = [];
         foreach ($this->referrers as $refFK) {
             if ($refFK->getTable()->isCrossRef()) {
-                $crossFK = new CrossForeignKeys($refFK, $this);
-                foreach ($refFK->getOtherFks() as $fk) {
-                    if ($fk->isAtLeastOneLocalPrimaryKeyIsRequired() &&
-                        $crossFK->isAtLeastOneLocalPrimaryKeyNotCovered($fk)) {
-                        $crossFK->addCrossForeignKey($fk);
+                // group all polymorphic fks
```

Comment:
Let's find a way to reduce the complexity of this class. It was already 2 nested foreach loops, both nested inside if statements, this has now increased a fair bit.

I'd also like to understand more about the performance impact of this change, and what cases users of Propel will need to be cautious about getting into.